src/SSRThread.hpp

```diff
@@ -17,12 +17,17 @@ class SSRThread : public QThread
                        std::string obfs_param,  //
                        std::string protocol,    //
                        std::string protocol_param);
-    void run() override;
     ~SSRThread() override;
   signals:
     void OnDataReady(quint64 dataUp, quint64 dataDown);
     void onSSRThreadLog(QString);
 
+protected:
+  void run() override;
+
+public slots:
```

Comment:
QThread does not have a restart mechanism. If you stop it manually, you will not be able to start it again. It may be a better idea to use destructor with unique_ptr.